project/src/main/java/com/google/sps/data/ProductSearchLibrary.java

```diff
@@ -275,17 +284,25 @@ private static void createProductSetItemAndAddToList(String productSetId, String
       productSetList.add(productSet);
     }
 
-    public static void deleteProduct(String productId)
-    throws IOException {
-        try (ProductSearchClient client = ProductSearchClient.create()) {
+  public static void deleteProduct(String productId)
+      throws IOException {
+        ProductSearchClient client = ProductSearchClient.create();
 
-            // Get the full path of the product.
-            String formattedName =
-                ProductSearchClient.formatProductName(projectId, computeRegion, productId);
+         // Get the full path of the product.
+        String formattedName =
+            ProductSearchClient.formatProductName(projectId, computeRegion, productId);
 
-            // Delete a product.
-            client.deleteProduct(formattedName);
-            System.out.println("Product deleted.");
-        }
-    }
+        // Delete a product.
+        client.deleteProduct(formattedName);
+  }
+
+  public static String getProductId(String productName){
+
+      String[] productNameSplit = productName.split("/");
+
+      int productNameArrayLength = productNameSplit.length;
+
+      return productNameSplit[productNameArrayLength - 1];
```

Comment:
You could skip a step here and just do `return productNameSplit[productNameSplit.length - 1];`